packages/instrumentation-pg/src/utils.ts

```diff
@@ -146,19 +154,39 @@ function getPort(port: number | undefined): number | undefined {
 }
 
 export function getSemanticAttributesFromConnection(
-  params: PgParsedConnectionParams
+  params: PgParsedConnectionParams,
+  semconvStability: SemconvStability
 ) {
-  return {
-    [SEMATTRS_DB_SYSTEM]: DBSYSTEMVALUES_POSTGRESQL,
-    [SEMATTRS_DB_NAME]: params.database, // required
-    [SEMATTRS_DB_CONNECTION_STRING]: getConnectionString(params), // required
-    [SEMATTRS_NET_PEER_NAME]: params.host, // required
-    [SEMATTRS_NET_PEER_PORT]: getPort(params.port),
-    [SEMATTRS_DB_USER]: params.user,
-  };
+  let attributes: Attributes = {};
+
+  if (semconvStability & SemconvStability.OLD) {
+    attributes = {
+      ...attributes,
+      [ATTR_DB_SYSTEM]: DB_SYSTEM_VALUE_POSTGRESQL,
+      [ATTR_DB_NAME]: params.database,
+      [ATTR_DB_CONNECTION_STRING]: getConnectionString(params),
+      [ATTR_DB_USER]: params.user,
+      [ATTR_NET_PEER_NAME]: params.host, // required
+      [ATTR_NET_PEER_PORT]: getPort(params.port),
+    };
+  }
+  if (semconvStability & SemconvStability.STABLE) {
+    attributes = {
+      ...attributes,
+      [ATTR_DB_SYSTEM_NAME]: DB_SYSTEM_VALUE_POSTGRESQL,
```

Comment:
nit:
```suggestion
      [ATTR_DB_SYSTEM_NAME]: DB_SYSTEM_NAME_VALUE_POSTGRESQL,
```

which would then need to be imported from `@opentelemetry/semantic-conventions`.